CHANGELOG.md

```diff
@@ -14,6 +14,14 @@ Release channels have their own copy of this changelog:
 
 <a name="edge-channel"></a>
 ## [2.1.0] - Unreleased
+* Breaking
```

Comment:
are we breaking semver here? 2.0.0 and 2.1.0 are to be compatible to adhere afaik

Reply:
I can certainly backport to v2.0 as well, if you think that's better.
If removals like this *have* to happen in the vX.0.0 release, we definitely need to provide more warning to engineers before branching than we did with v2.0.0.

Reply:
Also, fwiw, I had this PR up before the branch happened, but didn't receive reviews.

Reply:
I vote we proceed and backport to v2.0

Reply:
hmm... we should probably make deprecated api removal a blocker for major version branching in the future

i'm not opposed to a 2.0.0 bp. or at least i'm less opposed to that than removing deprecated api in a minor. @willhickey wdyt?

Reply:
> I vote we proceed and backport to v2.0

Okay, works for me. I will update the changelog in this PR to reflect that.
I will have ~3 follow-up PRs, which will need similar treatment.